syntax = "proto3";

package auth;

service AuthService {
  rpc Register (RegisterRequest) returns (AuthResponse);
  rpc Login (LoginRequest) returns (AuthResponse);
  rpc ValidateToken (ValidateRequest) returns (ValidateResponse);
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  bool success = 1;      
  string token = 2;
  string username = 3;
  string message = 4;
}

message ValidateRequest {
  string token = 1;
}

message ValidateResponse {
  bool is_valid = 1;
  string username = 2;
  string message = 3;
}